### PR TITLE
Implement deprecated setPosition and setOrientation methods for AudioListener #22898

### PR DESCRIPTION
**Pull request by snarasi6**
I have uncommented the two methods SetPosition and SetOrientation in AudioListener.webidl and have included the two methods in the AudioListener.rs file. I have some errors while handling the NotSupported error for the methods. Can you please review the code and help me out with the errors?

---

- [ ] `./mach build -d` does not report any errors
- [ ] `./mach test-tidy` does not report any errors
- [ ] These changes fix #___ (GitHub issue number if applicable)

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because ___

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/23279)

### CODE REVIEW

**Comment by jdm** [comment]
It looks to me like the error messages you're getting describe the problem and the solution:
```
error[E0053]: method `SetOrientation` has an incompatible type for trait
    --> components/script/dom/audiolistener.rs:195:133
     |
195  |     fn SetOrientation(&self, x: Finite<f32>, y: Finite<f32>, z: Finite<f32>, xUp: Finite<f32>, yUp: Finite<f32>, zUp: Finite<f32>)->() {
     |                                                                                                                                     ^^ expected enum `std::result::Result`, found ()
     | 
    ::: /home/travis/build/servo/servo/target/debug/build/script-05686a0f060e3ec8/out/Bindings/AudioListenerBinding.rs:1065:135
     |
1065 |     fn SetOrientation(&self, x: Finite<f32>, y: Finite<f32>, z: Finite<f32>, xUp: Finite<f32>, yUp: Finite<f32>, zUp: Finite<f32>) -> Fallible<()>;
     |                                                                                                                                       ------------ type in trait
     |
     = note: expected type `fn(&dom::audiolistener::AudioListener, dom::bindings::num::Finite<f32>, dom::bindings::num::Finite<f32>, dom::bindings::num::Finite<f32>, dom::bindings::num::Finite<f32>, dom::bindings::num::Finite<f32>, dom::bindings::num::Finite<f32>) -> std::result::Result<(), dom::bindings::error::Error>`
                found type `fn(&dom::audiolistener::AudioListener, dom::bindings::num::Finite<f32>, dom::bindings::num::Finite<f32>, dom::bindings::num::Finite<f32>, dom::bindings::num::Finite<f32>, dom::bindings::num::Finite<f32>, dom::bindings::num::Finite<f32>)`
``

**Comment by jdm** [comment]
That being said, returning `Err(NotSupported)` from the methods is not what the spec says. You will want to return `Ok(())` when there is no automation curve present.

**Comment by Akhilesh1996** [comment]
Josh, I have made the changes and squashed commits. The build is successful but /mach test-tidy reports ./components/script/dom/audiolistener.rs:190: method declared in webidl is missing a comment with a specification link. But I have added a comment in the audiolistener.rs file for both the methods. I have run the tests in the webaudio. folder and updated results and reverted results for webaudio/the-audio-api/the-iirfilternode-interface/iirfilter.html.  Can you check if what has been done till now is right and if I should make more changes? 
 
 * [X]  `./mach build -d` does not report any errors
 * [ ]  `./mach test-tidy` does not report any errors
 * [X]  These changes fix #22898 (GitHub issue number if applicable)
 
 * [ ]  There are tests for these changes OR
 * [ ]  These changes do not require tests because ___
 
 This change is [Reviewable](https://reviewable.io/reviews/servo/servo/23279)

File: components/script/dom/audiolistener.rs [hunk 1]
```diff
@@ -181,4 +186,35 @@ impl AudioListenerMethods for AudioListener {
...
+        y: Finite<f32>,
+        z: Finite<f32>,
+        xUp: Finite<f32>,
+        zUp: Finite<f32>,
+    ) -> Fallible<DomRoot<AudioListener>> {
+        self.forward_x.SetValue(x);
+        self.forward_y.SetValue(y);
+        self.forward_z.SetValue(z);
+        self.up_x.SetValue(xUp);
+        self.up_y.SetValue(yUp);
```
Comment (jdm) [inline components/script/dom/audiolistener.rs]: yUp doesn't exist?

**Review by jdm — CHANGES_REQUESTED** [review]
This looks correct but doesn't compile right now.

**Comment by jdm** [comment]
Also if you add a space after the // the tidy should be happy.

File: components/script/dom/audiolistener.rs (hunk 1 above)
Comment (Akhilesh1996) [reply components/script/dom/audiolistener.rs]: Thanks for pointing out, I have fixed It and addressed the errors from ./mach test-tidy. Can you check now?

**Pull request merged**